crates/fuel-streams-core/src/nats/store.rs

```diff
@@ -0,0 +1,90 @@
+use std::fmt::Debug;
+
+use async_nats::jetstream::{
+    consumer::{FromConsumer, IntoConsumerConfig},
+    kv::Watch,
+};
+use async_trait::async_trait;
+use fuel_streams_macros::subject::IntoSubject;
+
+use super::{NatsClient, NatsNamespace, StoreError};
+use crate::nats::{types::*, NatsError};
+
+#[async_trait]
+pub trait Storable: Debug + Clone + serde::Serialize {
+    const STORE: &'static str;
+
+    /// This is temporary until we have the data parser
+    fn store_encode(&self) -> Result<Vec<u8>, StoreError> {
+        bincode::serialize(self).map_err(StoreError::SerializationFailed)
+    }
+
+    /// This is temporary until we have the data parser
+    fn store_decode<'a>(val: &'a [u8]) -> Result<Self, StoreError>
+    where
+        Self: serde::Deserialize<'a>,
+    {
+        bincode::deserialize(val).map_err(StoreError::SerializationFailed)
+    }
+
+    async fn create_store(
```

Comment:
This is fine (`create_kv_store`), since for a `fuel-core-type` to be a `Streamable` we need a nice helper function to create a KV store, but we could go further by taking it away from this public trait and making private.